### PR TITLE
T 10508

### PR DESCRIPTION
**Pull request by schaffe**

### CODE REVIEW

File: src/com/backendless/Backendless.java [hunk 1]
```diff
@@ -126,24 +120,20 @@ public static void initApp( Object context, final String applicationId, final St
...
-    backendlessInitService.initService( context, new IServiceCreatedCallback()
-    {
-      @Override
-      public void onServiceConnected( IBackendlessService result )
-      {
-        backendlessService = result;
-        backendlessService.setAuthKeys( applicationId, secretKey, version );
-      }
-    } );
+    prefs.onCreate( context );
```
Comment (markpiller) [inline src/com/backendless/Backendless.java]: what if this is not android. Does this line work as expected?

File: src/com/backendless/Messaging.java [hunk 2]
```diff
@@ -107,47 +107,43 @@ static Messaging getInstance()
     return instance;
   }
 
-  public void registerDevice( String GCMSenderID )
+  public void registerDevice( Context context, String GCMSenderID )
```
Comment (markpiller) [inline src/com/backendless/Messaging.java]: this breaks backward compatibility

File: src/com/backendless/Messaging.java [hunk 3]
```diff
@@ -107,47 +107,43 @@ static Messaging getInstance()
...
   {
-    registerDevice( GCMSenderID, "" );
+    registerDevice( context, GCMSenderID, "" );
   }
 
-  public void registerDevice( String GCMSenderID, String channel )
+  public void registerDevice( Context context, String GCMSenderID, String channel )
```
Comment (markpiller) [inline src/com/backendless/Messaging.java]: this breaks backward compatibility

File: src/com/backendless/Messaging.java [hunk 4]
```diff
@@ -107,47 +107,43 @@ static Messaging getInstance()
...
-  public void registerDevice( String GCMSenderID, List<String> channels, Date expiration )
+  public void registerDevice( Context context, String GCMSenderID, String channel, AsyncCallback<Void> callback )
   {
-    registerDevice( GCMSenderID, channels, expiration, null );
+    registerDevice( context, GCMSenderID, (channel == null || channel.equals( "" )) ? null : Arrays.asList( channel ), null, callback );
   }
 
-  public void registerDevice( final String GCMSenderID, final List<String> channels, final Date expiration,
-                              final AsyncCallback<Void> callback )
+  public void registerDevice( Context context, String GCMSenderID, List<String> channels, Date expiration )
```
Comment (markpiller) [inline src/com/backendless/Messaging.java]: this breaks backward compatibility

File: src/com/backendless/Messaging.java [hunk 5]
```diff
@@ -175,11 +171,6 @@ protected void onPostExecute( RuntimeException result )
     }.execute();
   }
 
-  public void registerDevice( String GCMSenderID, AsyncCallback<Void> callback )
```
Comment (markpiller) [inline src/com/backendless/Messaging.java]: this breaks backward compatibility

File: src/com/backendless/Messaging.java [hunk 6]
```diff
@@ -263,31 +254,20 @@ public void handleFault( BackendlessFault fault )
     }
   }
 
-  public void unregisterDevice()
+  public void unregisterDevice(Context context)
```
Comment (markpiller) [inline src/com/backendless/Messaging.java]: this breaks backward compatibility

File: src/com/backendless/Backendless.java (hunk 1 above)
Comment (schaffe) [inline src/com/backendless/Backendless.java]: That`s just a name

**Pull request merged**